lib/clicksign/api/requests.rb

```diff
@@ -2,23 +2,37 @@ module Clicksign
   module API
     module Requests
       def post(request_path, body, token)
-        conn.post do |req|
+        response = conn.post do |req|
           req.url request_path, { access_token: Clicksign::API.credentials[token] }
           req.headers['Content-Type'] = 'application/json'
           req.body = body.to_json
         end
+        
+        parse(response)
       end
 
       def get(request_path, token)
-        conn.get do |req|
+        response = conn.get do |req|
           req.url request_path, { access_token: Clicksign::API.credentials[token] }
           req.headers['Content-Type'] = 'application/json'
         end
+        
+        parse(response)
       end
+      
+      private
 
       def conn
         @conn ||= Faraday.new(url: Clicksign::API.url)
       end
+      
+      def parse(response)
+        if !response.body.empty?
+          JSON.parse(response.body, symbolize_keys: true).merge(status: response.status)
```

Comment:
Sabe o que seria legal aqui? Nice to have total, não precisa fazer.
Eu colocaria uma classe `Response` que respondesse para `.success?` e cujo corpo fosse devidamente tratado, para devolver um hash sempre que possível e `nil` quando o corpo da resposta fosse uma string vazia.

Reply:
e aí só chamaria a classe `Response.success` no `Requests.rb`, e não precisaria tratar aí dentro do Resquests, separando a responsabilidade de cada coisa. Saquei a ideia Chico. Valeu... Vou ver o que posso fazer pra implementar isso com o time, porque estamos com lead time lascado de apertado nessa task e o teste com stakeholder será amanhã. Mas isso tbm deve ser rápido, acho que vou trocar ideia com eles e vemos isso. Boa sugestão 👍🏼